engine/runtime/src/main/java/org/enso/interpreter/runtime/TruffleCompilerContext.java

```diff
@@ -276,7 +275,7 @@ public void shutdown(boolean waitForPendingJobCompletion) {
 
   private final class ModuleUpdater implements Updater, AutoCloseable {
     private final Module module;
-    private BindingsMap map;
+    private BindingsMap[] map;
```

Comment:
Why did this become an array?

I can see only the 0-th element is ever accessed.

Reply:
To allow setting `null` value. I need to differentiate between `null` (e.g. no change) and `null` being set. Hence one argument array.

Reply:
This is just repeating the same (undocumented) pattern as used on the next line with `org.enso.compiler.core.ir.Module[] ir`.